x-pack/plugin/logstash/src/main/java/org/elasticsearch/xpack/logstash/action/TransportGetPipelineAction.java

```
@@ -0,0 +1,143 @@
+                        if (response != null && response.getScrollId() != null) {
+                            ClearScrollRequest clearScrollRequest = new ClearScrollRequest();
+                            clearScrollRequest.addScrollId(response.getScrollId());
+                            client.clearScroll(
+                                clearScrollRequest,
+                                ActionListener.wrap(
+                                    (r) -> {},
+                                    e -> logger.warn(
+                                        new ParameterizedMessage("clear scroll failed for scroll id [{}]", response.getScrollId()),
+                                        e
+                                    )
+                                )
+                            );
+                        }
+                    };
+                    handleSearchResponse(searchResponse, pipelineSources, clearScroll, listener);
+                }, listener::onFailure));
+        } else if (request.ids().size() == 1) {
+            client.prepareGet(".logstash", request.ids().get(0)).setFetchSource(true).execute(ActionListener.wrap(response -> {
+                if (response.isExists()) {
+                    listener.onResponse(new GetPipelineResponse(Map.of(response.getId(), response.getSourceAsBytesRef())));
+                } else {
+                    listener.onResponse(new GetPipelineResponse(Map.of()));
+                }
+            }, listener::onFailure));
+        } else {
+            client.prepareMultiGet()
+                .addIds(".logstash", request.ids())
+                .execute(
+                    ActionListener.wrap(
+                        mGetResponse -> listener.onResponse(
+                            new GetPipelineResponse(
+                                Arrays.stream(mGetResponse.getResponses())
+                                    .filter(itemResponse -> itemResponse.isFailed() == false)
```

I'm not sure what to do instead, but if some of the GETs fail, do we really just want to silently swallow that? Should we at least log something here?